yapapi/ctx.py

```diff
@@ -351,63 +122,64 @@ def _payment_model(self) -> ComLinear:
 
         return self.__payment_model
 
-    def __prepare(self):
-        if not self._started and self._implicit_init:
-            self.deploy()
-            self.start()
-            self._started = True
+    def new_script(self):
+        """Create an instance of `Script` attached to this `WorkContext` instance.
 
-    def begin(self):
-        pass
+        This is equivalent to calling `Script(work_context)`. This method is intended to provide a
+        direct link between the two object instances.
+        """
+        return Script(self)
 
+    @deprecated(version="0.7.0", reason="please use a Script object via WorkContext.new_script")
     def deploy(self):
         """Schedule a Deploy command."""
-        self._implicit_init = False
-        self._pending_steps.append(_Deploy())
+        self.__script.deploy()
 
+    @deprecated(version="0.7.0", reason="please use a Script object via WorkContext.new_script")
     def start(self, *args: str):
         """Schedule a Start command."""
-        self._implicit_init = False
-        self._pending_steps.append(_Start(*args))
+        self.__script.start(*args)
 
+    @deprecated(version="0.7.0", reason="please use a Script object via WorkContext.new_script")
     def terminate(self):
         """Schedule a Terminate command."""
-        self._pending_steps.append(_Terminate())
+        self.__script.terminate()
 
+    @deprecated(version="0.7.0", reason="please use a Script object via WorkContext.new_script")
     def send_json(self, json_path: str, data: dict):
         """Schedule sending JSON data to the provider.
 
         :param json_path: remote (provider) path
         :param data: dictionary representing JSON data
         :return: None
         """
-        self.__prepare()
-        self._pending_steps.append(_SendJson(self._storage, data, json_path))
+        self.__script.send_json(data, json_path)
```

Comment:
Is there any reason why `WorkContext.send_json` and `Script.send_json` have different order of arguments?